### PR TITLE
rename accelerator_backend -> accelerator

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Follow up to #5743 
Shorter, cleaner. There is no confusion that we are using the accelerator.

It's backward compatible, we have a property that redirects accelerator_backend to accelerator :) 

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pl_examples/basic_examples/conv_sequential_example.py [hunk 1]
```diff
@@ -222,6 +222,6 @@ def instantiate_datamodule(args):
     trainer.fit(model, cifar10_dm)
     trainer.test(model, datamodule=cifar10_dm)
 
-    if trainer.accelerator_backend.rpc_enabled:
+    if trainer.accelerator.rpc_enabled:
```
Comment (Borda) [inline pl_examples/basic_examples/conv_sequential_example.py:225]: just a side note, wont this confuse user with Trainer argument which is string?

Comment (awaelchli) [reply pl_examples/basic_examples/conv_sequential_example.py:225]: good point, I have not thought about this. 
On the other hand, we never store the string input anywhere, so the user has to go through distrib_type.
@justusschock what is your preference?

Comment (tchaton) [reply pl_examples/basic_examples/conv_sequential_example.py:225]: Hey both, I think `accelerator` is better. Most users might not use the trainer directly and even less the accelerator.

However, we should add a depreciation there for `accelerator_backend` property @awaelchli.

Comment (justusschock) [reply pl_examples/basic_examples/conv_sequential_example.py:225]: I also think we should rename it and add a property linking to it for the old name with raising a depreciation warning as @tchaton mentioned.

**Review by justusschock — APPROVED** [review]
Like it. Please just add the property for backwards compatibility

**Review by SeanNaren — APPROVED** [review]
I love this, was one of my first confusions when dealing with lightning :)

**Comment by awaelchli** [comment]
@justusschock 👍 It is already there, from the accelerator refactor.

File: pl_examples/basic_examples/conv_sequential_example.py (hunk 1 above)
Comment (awaelchli) [reply pl_examples/basic_examples/conv_sequential_example.py:225]: ok, will add that warning

**Review by carmocca — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Comment by Borda** [comment]
@tchaton mind review?

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**